openpype/hosts/nuke/plugins/publish/collect_writes.py

```diff
@@ -48,113 +33,134 @@ def process(self, instance):
             )
             return
 
-        instance.data["writeNode"] = write_node
-        self.log.debug("checking instance: {}".format(instance))
+        # get colorspace and add to version data
+        colorspace = napi.get_colorspace_from_node(write_node)
 
-        # Determine defined file type
-        ext = write_node["file_type"].value()
+        if render_target == "frames":
+            self._set_existing_files_data(instance, colorspace)
 
-        # Get frame range
-        handle_start = instance.context.data["handleStart"]
-        handle_end = instance.context.data["handleEnd"]
+        elif render_target == "frames_farm":
+            collected_frames = self._set_existing_files_data(
+                instance, colorspace)
+
+            self._set_expected_files(instance, collected_frames)
+
+            self._add_farm_instance_data(instance)
+
+        elif render_target == "farm":
+            self._add_farm_instance_data(instance)
+
+        # set additional instance data
+        self._set_additional_instance_data(instance, render_target, colorspace)
+
+    def _set_existing_files_data(self, instance, colorspace):
```

Comment:
should we move this function into something more generic that we can reuse elsewhere?

Reply:
After further analysis of the code, I have not identified any instances where abstraction could be utilized elsewhere.

Reply:
maybe this one is quite specific to the Nuke write node but similarly to my other comment https://github.com/ynput/OpenPype/pull/5409/files#r1286334363 there will be other plugin developers that would start implementing the same logic for setting existing files on representations. For example, the TrayPublisher farm support that I added https://github.com/ynput/OpenPype/pull/5195/files#diff-570c6ac4c5667daf43a0e9404fa8973fa092c74d94c8caa2ceb670c225ff77c5R128 I kind of hacked the framework so the `expectedFiles` attribute would turn into `files` on the `submit_publish_job.py:prepare_representations` because there was no existing common functions for doing exactly that. For example, look at how many duplicates of `generate_expected_files` we have in the codebase: https://github.com/search?q=repo%3Aynput%2FOpenPype+%22generate_expected_files%22&type=code , that's what I'd like to avoid

Reply:
I agree that we need to provide an abstraction for existing frames. However, let's address this in a separate pull request.